src/warnings.ts

```diff
@@ -28,6 +29,10 @@ export function warn(code: string, message: string) {
   if (isBrowser()) {
     console.warn(message);
   } else {
-    process.emitWarning(message);
+    if (type !== 'undefined') {
+      process.emitWarning(message, type);
+    } else {
+      process.emitWarning(message);
+    }
```

Comment:
Could you add a unit test here? 
https://github.com/googleapis/gax-nodejs/blob/master/test/unit/warning.ts